### ISSUE TITLE
jspm install munges sass.worker.js file at install

### ISSUE THREAD

**Issue by theefer** [jspm/jspm-cli#906]
When installing the `npm:sass.js` package, JSPM seems to be rewriting the file at `dist/sass.worker.js` by (1) expanding the lines (previously minimised), causing the file to bloat in size, and (2) by injected dependencies into a form of CommonJS IIFE, which breaks the script when run as a Web Worker in the browser.

See file installed via JSPM:

```
$ du -sch jspm_packages/npm/sass.js@0.9.2/dist/sass.worker.js 
7.7M    jspm_packages/npm/sass.js@0.9.2/dist/sass.worker.js

$ head -n 5 jspm_packages/npm/sass.js@0.9.2/dist/sass.worker.js 
/* */ 
(function(Buffer, process) {
  var Module = {onRuntimeInitialized: function() {
      'use strict';
      Sass._ready();

$ tail -n 5 jspm_packages/npm/sass.js@0.9.2/dist/sass.worker.js 
      });
    }
    method.apply(Sass, (event.data.args || []).concat([done]));
  };
})(require("buffer").Buffer, require("process"));
```

vs file installed via npm:

```
$ du -sch node_modules/sass.js/dist/sass.worker.js 
2.6M    node_modules/sass.js/dist/sass.worker.js

$ head -n 10 node_modules/sass.js/dist/sass.worker.js  
/*! sass.js - v0.9.2 (e355668) - built 2015-06-14
  providing libsass 3.2.5 (0e6b4a2)
  via emscripten 1.33.1 (ae2f801)
 */
/*global Sass*/
var Module = {
  onRuntimeInitialized: function() {
    'use strict';
    Sass._ready();
  }

$ tail -n 10 node_modules/sass.js/dist/sass.worker.js  

  if (!method) {
    return done({
      line: 0,
      message: 'Unknown command ' + event.action
    });
  }

  method.apply(Sass, (event.data.args || []).concat([done]));
};
```

**Update: corrected head/tail for npm installed sass.js**

This has been reproduced both with JSPM 0.15 and 0.16.

This problem was found when working on a [Sass plugin for SystemJS](https://github.com/theefer/plugin-sass).

**Comment by theefer** [comment]
Oops I'd mis-copied the head/tail of the npm-installed sass.js, fixed above now.

**Comment by guybedford** [comment]
How does the above break in a web worker? These conversions are part of making node modules work in jspm, so happy to add a fix to them, but the rewriting is important.

**Comment by theefer** [comment]
So my plugin loads the `sass.js` file, which is the public interface to the sass.js library. However as far as I can see, the `sass.worker.js` file is never loaded by `sass.js`, it is instead run independently in a Web Worker (using `new Worker(urlToSassWorkerJs)`). When the browser tries to run the JSPM-altered  `sass.worker.js`, it fails on the last line:

`})(require("buffer").Buffer, require("process"));`

because of course there is no `window.require` in the isolated context of a Web Worker.

In other words, it feels like we should be able to make JSPM completely ignore that file, because it will never actually get loaded in a JSPM context. Is that the right approach, and is that possible?

**Comment by guybedford** [comment]
We don't currently have a way to disable the conversion process for just some files. Currently the suggestion is to load them as separate packages (installing worker version from github repo eg). But something along the lines of selective conversion disabling sounds like it might be the solution in the long term.

**Comment by guybedford** [comment]
Note that in future, no static file operations will be done as part of the node conversion process - it will all be meta-configuration-based (when 0.17 finally lands). So that this issue would no longer be a problem. Introducing alternative configuration here is out of scope unfortunately. Let me know if you have further questions on this at all.

**Comment by theefer** [comment]
Do you have any example of another plugin or package that loads some files from GitHub/NPM separately?

**Comment by guybedford** [comment]
So there is a way you can do this with an override actually, using a shim:

```
  jspm install npm:sass.js -o "{shim: { 'dist/sass.worker': {} } }"
```

**Comment by theefer** [comment]
Done in https://github.com/jspm/registry/pull/509, and it worked, which means plugin-sass now works out of the box!

Thanks a lot @guybedford!

Next step will be to fix the bundle process.

/cc @OliverJAsh @kaelig

**Comment by guybedford** [comment]
Great, just let me know if I can help further.

**Comment by theefer** [comment]
@guybedford All good here, I'm just waiting for the next jspm release to test your fix for https://github.com/theefer/plugin-sass/issues/2#issuecomment-125581661, which should fix the bundle phase for the Sass plugin. :grinning:
